File: test-suite/src/test/java/io/micronaut/docs/inject/intro/V8Engine.java
package io.micronaut.docs.inject.intro;

import javax.inject.Singleton;

// tag::class[]
@Singleton// <2>
public class V8Engine implements Engine {
    public String start() {
        return "Starting V8";
    }

    public int getCylinders() {
        return cylinders;
    }

    public void setCylinders(int cylinders) {
        this.cylinders = cylinders;
    }

    private int cylinders = 8;
}
// end::class[]

File: test-suite-scala/src/test/scala/io/micronaut/docs/inject/intro/V8Engine.scala
package io.micronaut.docs.inject.intro

import javax.inject.Singleton

@Singleton class V8Engine extends Engine {

  override def start: String =  "Starting V8"

  override def getCylinders: Int = cylinders

  def setCylinders (cylinders: Int) {
    this.cylinders = cylinders
  }

  private var cylinders: Int = 8
}
